_posts/2018-08-28-announcing-riff-0-1-2-on-Knative.md

```diff
@@ -0,0 +1,197 @@
+  --artifact square.js \
+  --image $DOCKER_ID/node-fun-square:v1 \
+  --wait
+```
+```
+Error: function creation failed: RevisionMissing: Configuration "square" does not have any ready Revision.; Revision creation failed with message: "Internal error occurred: admission webhook \"webhook.build.knative.dev\" denied the request: mutation failed: serviceaccounts \"riff-build\" not found".
+```
+
+Or, if the specified artifact does not exist, `--verbose` will help you figure this out.
+
+```
+riff function create node square \
+  --git-repo https://github.com/trisberg/node-fun-square.git \
+  --artifact squarexxx.js \
+  --image $DOCKER_ID/node-fun-square \
+  --verbose
+```
+```
+...
+default/square-00001-c2k97[build-step-build-and-push]: time="2018-08-29T16:34:13Z" level=error msg="lstat /workspace/squarexxx.js: no such file or directory"
+Error: function creation failed: RevisionMissing: Configuration "square" does not have any ready Revision.; Revision "square-00001" failed with message: "build step \"build-step-build-and-push\" exited with code 1 (image: \"docker-pullable://gcr.io/kaniko-project/executor@sha256:501056bf52f3a96f151ccbeb028715330d5d5aa6647e7572ce6c6c55f91ab374\"); for logs run: kubectl -n default logs square-00001-c2k97 -c build-step-build-and-push".
+```
+
+## function chaining over channels
+
+[Knative/eventing](https://github.com/knative/eventing) now includes support for subscriptions to [send function responses](https://github.com/knative/eventing/pull/325) to another channel. This feature can be used through the riff CLI by specifying the output channel with `--output` or `-o` when creating a subscription.
+
+### 1. create a simple hello function
+Start with 2 files in a directory.
+
+#### hello.js  
+```js
+module.exports = x => {
+  var out = 'hello ' + x
+  console.log(out)
+  return out
+}
+```
+
+#### Dockerfile  
+```dockerfile
+FROM projectriff/node-function-invoker:0.0.8
+ENV FUNCTION_URI /functions/hello.js
+ADD hello.js ${FUNCTION_URI}
+```
+
+Now build the function into a container image, and use it create a Knative Service. The `dev.local` prefix tells Knative to use the image from the local docker daemon instead of pulling an image from a remote container registry.
+```sh
+docker build -t dev.local/hello:v1 .
+riff service create hello --image dev.local/hello:v1
+```
+> To make this work in a hosted environment like GKE, change the names of the images to prefix them with your own registry, and push the images there before calling riff service create.
+
+Using a tool like kail makes it easy to watch the function container log in a separate terminal window.
+```sh
+kail -d hello-00001-deployment -c user-container
+```
+
+When you invoke hello, kail relays the container log.
+```sh
+riff service invoke hello -- -HContent-Type:text/plain '-w\n' -d hello
+```
+
+#### kail output
+```
+...[user-container]: hello hello
+```
+
+### 2. create a square function
```

Comment:
We already have the square function from Thomas (and it is used in this very blog post). Why not re-use it?

Reply:
yes - good idea - i will figure out a way to re-use it to reduce the length of the post - the logging for square isn't essential in this case.